Make the id editable

guglo opened the pull request:
This checkin includes:

1. Removed name field in edit server modal and model.
2. Made the id editable in edit server modal.
3. Added unique id validation. 
4. Updated to make ID not duplicate when edit server. 
5. Fixed to not allow  duplicate of server added manually.
6. Fixed problem where the last valid input can turn on the add more or save
   button while the previous invalid inputs are ingnored.
7. Cleanup for unused cpu and ram item in the server data
8. Fixed issue when user updates in server role summary but the leftside
    servers are not updated accordingly.
9. Fixed issue where the accordion table has part of gray and part of white
   when overflow
10. Fixed issue of duplicates servers when import the same server (id) multiple times. 
11. Turn the accordion red and server items red when duplicates found in the rightside model table and the next button disabled.

Commit:
Make the id editable - using uid WIP

This checkin includes:

1. Removed name field in edit server modal and model
2. Made the id editable in edit server modal
3. Added unique id validation to make sure no duplicate id updated/added
4. Fixed to not allow  duplicate of server added manually
5. Fixed problem where the last valid input can turn on the add more or save
   button while the previous invalid inputs are ingnored.
6. Cleanup for unused cpu and ram item in the server data
7. Fixed issue when user updates in server role summary but the leftside
    servers are not updated accordingly.
8. Fixed issue where the accordion table has part of gray and part of white
   when overflow

Note:
To make this change smaller, will address existing problems separately
1. import duplicates in csv

guglo commented:
This ui changes goes with backend https://github.com/ArdanaCLM/ardana-installer-server/pull/20

Commit:
Fixed import server duplicates issue

Commit:
Fixed delete issue to use uid

Commit:
Fixed a couple of import issues

Commit:
minor lints fix

Commit:
Minor cleanup

jeremy-moffitt reviewed (approved):
looks ok to me. I tried it out and it worked as expected

Commit:
Fixed issues based on Mohan's feedback

Fixed issue when user drag the example controller but drop back to model
table.
Added configured tab for the leftside auto discovered servers so it
is clear which is discovered and which could be configured and changed in
case user changes hostname, ip or mac address.

Commit:
Make sure required field to be filled in manual add server
